test/index.spec.ts

```diff
@@ -137,452 +137,540 @@ test('encryptor:encrypt & decrypt with wrong password', async ({ page }) => {
+
+      const decryptResult = await page.evaluate(
+        async (args) => {
+          return await window.encryptor.decrypt(
+            args.password,
+            JSON.stringify(args.testEncryptedData),
+          );
+        },
+        { password, testEncryptedData },
       );
-    },
-    { password, sampleEncryptedData },
-  );
 
-  const decryptWithDetailResult = await page.evaluate(
-    async (args) => {
-      return await window.encryptor.decryptWithDetail(
-        args.password,
-        JSON.stringify(args.sampleEncryptedData),
+      const decryptWithDetailResult = await page.evaluate(
+        async (args) => {
+          return await window.encryptor.decryptWithDetail(
+            args.password,
+            JSON.stringify(args.testEncryptedData),
+          );
+        },
+        { password, testEncryptedData },
       );
-    },
-    { password, sampleEncryptedData },
-  );
-
-  expect(JSON.stringify(decryptResult)).toStrictEqual(
-    JSON.stringify(decryptWithDetailResult.vault),
-  );
-  expect(Object.keys(decryptWithDetailResult).length).toBe(3);
-  expect(typeof decryptWithDetailResult.exportedKeyString).toStrictEqual(
-    'string',
-  );
-});
-
-test('encryptor:encrypt using key then decrypt', async ({ page }) => {
-  const password = 'a sample passw0rd';
-  const data = { foo: 'data to encrypt' };
-  const salt = await page.evaluate(() => window.encryptor.generateSalt());
 
-  const encryptedData = await page.evaluate(
-    async (args) => {
-      const key = await window.encryptor.keyFromPassword(
-        args.password,
-        args.salt,
+      expect(JSON.stringify(decryptResult)).toStrictEqual(
+        JSON.stringify(decryptWithDetailResult.vault),
       );
-      return await window.encryptor.encryptWithKey(key, args.data);
-    },
-    { data, password, salt },
-  );
-  expect(Object.keys(encryptedData).sort()).toStrictEqual(['data', 'iv']);
-
-  const encryptedString = JSON.stringify(
-    Object.assign({}, encryptedData, { salt }),
-  );
-  const decryptedData = await page.evaluate(
-    async (args) =>
-      await window.encryptor.decrypt(args.password, args.encryptedString),
-    { encryptedString, password },
-  );
-
-  expect(decryptedData).toStrictEqual(data);
-});
-
-test('encryptor:encrypt using key then decrypt using wrong password', async ({
-  page,
-}) => {
-  const password = 'a sample passw0rd';
-  const wrongPassword = 'a wrong password';
-  const data = { foo: 'data to encrypt' };
-  const salt = await page.evaluate(() => window.encryptor.generateSalt());
-
-  const encryptedData = await page.evaluate(
-    async (args) => {
-      const key = await window.encryptor.keyFromPassword(
-        args.password,
-        args.salt,
+      expect(Object.keys(decryptWithDetailResult).length).toBe(3);
+      expect(typeof decryptWithDetailResult.exportedKeyString).toStrictEqual(
+        'string',
+      );
+    });
+
+    test('encryptor:encrypt using key then decrypt', async ({ page }) => {
```

Comment:
This test doesn't rely on either `sampleEncryptedData` or `oldSampleEncryptedData`. Should it be inside the loop?